firefox-ios/firefox-ios-tests/Tests/StorageTests/RustAutofillTests.swift

```diff
@@ -35,229 +35,200 @@ class RustAutofillTests: XCTestCase {
-
-        addCreditCard { creditCard, err in
-            XCTAssertNotNil(creditCard)
-            XCTAssertNil(err)
-            expectationAddCard.fulfill()
-
-            self.autofill.getCreditCard(id: creditCard!.guid) { card, error in
-                XCTAssertNotNil(card)
-                XCTAssertNil(err)
-                XCTAssertEqual(creditCard!.guid, card!.guid)
-                expectationGetCard.fulfill()
-
-                self.autofill.deleteCreditCard(id: card!.guid) { success, err in
-                    XCTAssert(success)
-                    XCTAssertNil(err)
-                    expectationDeleteCard.fulfill()
-
-                    self.autofill.getCreditCard(id: creditCard!.guid) { deletedCreditCard, error in
-                        XCTAssertNil(deletedCreditCard)
-                        XCTAssertNotNil(error)
-
-                        let expectedError =
-                        "NoSuchRecord(guid: \"\(creditCard!.guid)\")"
-                        XCTAssertEqual(expectedError, "\(error!)")
-                        expectationCheckDeleteCard.fulfill()
-                    }
-                }
-            }
+    func testListAllAddressesEmpty() async throws {
+        let addresses = try await listAllAddresses()
+
+        XCTAssertEqual(addresses.count, 0, "Addresses count should be 0 for an empty list")
+    }
+
+    func testUpdateCreditCard() async throws {
+        let creditCard = try await addCreditCard()
+        let card = try await getCreditCard(id: creditCard.guid)
+        let updatedCreditCard = UnencryptedCreditCardFields(ccName: creditCard.ccName,
+                                                            ccNumber: creditCard.ccNumberEnc,
+                                                            ccNumberLast4: creditCard.ccNumberLast4,
+                                                            ccExpMonth: creditCard.ccExpMonth,
+                                                            ccExpYear: Int64(2028),
+                                                            ccType: creditCard.ccType)
+        let result = try await updateCreditCard(id: creditCard.guid, creditCard: updatedCreditCard)
+        let updatedCardVal = try await getCreditCard(id: creditCard.guid)
+
+        XCTAssertEqual(creditCard.guid, card.guid)
+        XCTAssertTrue(result)
+        XCTAssertEqual(updatedCardVal.ccExpYear, updatedCreditCard.ccExpYear)
+    }
+
+    func testDeleteCreditCard() async throws {
+        let creditCard = try await addCreditCard()
+        let retrievedCreditCard = try await getCreditCard(id: creditCard.guid)
+        let deleteCreditCardResult = try await deleteCreditCard(id: retrievedCreditCard.guid)
+
+        do {
+            _ = try await getCreditCard(id: creditCard.guid)
+        } catch {
+            let expectedError = "NoSuchRecord(guid: \"\(creditCard.guid)\")"
+            XCTAssertEqual(expectedError, "\(error)")
```

Comment:
Here, I used a `do catch` block to check the error message when call the `getCreditCard` function. If I don't use `do catch` block, the test always fails.

Reply:
Sure. It makes sense. Previously I kept the error message assertion, thinking it is important. But for this test, it appears that only the nil assertion is enough.